src/main/java/edu/harvard/iq/dataverse/api/Files.java

```diff
@@ -818,4 +829,50 @@ public Response getExternalToolFMParams(@Context ContainerRequestContext crc, @P
     public Response getFixityAlgorithm() {
         return ok(systemConfig.getFileFixityChecksumAlgorithm().toString());
     }
+
+    @GET
+    @AuthRequired
+    @Path("{id}/guestbookResponses/count")
+    public Response getCountGuestbookResponses(@Context ContainerRequestContext crc, @PathParam("id") String dataFileId) {
+        return response(req -> {
+            DataFile dataFile = execCommand(new GetDataFileCommand(req, findDataFileOrDie(dataFileId)));
+            return ok(guestbookResponseService.getCountGuestbookResponsesByDataFileId(dataFile.getId()).toString());
+        }, getRequestUser(crc));
+    }
+
+    @GET
+    @AuthRequired
+    @Path("{id}/canBeDownloaded")
+    public Response canFileBeDownloaded(@Context ContainerRequestContext crc, @PathParam("id") String dataFileId) {
+        DataFile dataFile;
+        try {
+            dataFile = findDataFileOrDie(dataFileId);
+        } catch (WrappedResponse wr) {
+            return wr.getResponse();
+        }
+        return ok(fileDownloadServiceBean.canDownloadFile(getRequestUser(crc), dataFile.getFileMetadata()));
+    }
+
```

Comment:
Maybe a little Javadoc here about what thumbnail class is?

And we'll be documenting this for end users. We'll need to explain what thumbnailClass is in the API Guide. Is this a user-friendly enough term. Thumbnail image? Not sure. 🤷 Let's not confuse the API users.

Reply:
Maybe "grouping"? That's really what we're doing:

<img width="146" alt="Screenshot 2023-07-27 at 12 27 06 PM" src="https://github.com/IQSS/dataverse/assets/21006/60328d01-921e-44ca-b08c-9d74b9eaa301">

Reply:
Or just group? thumbnailGroup?

Reply:
I think we may be mixing concepts here. What we are grouping there are file type facets. Take a look at this comment: https://github.com/IQSS/dataverse/blob/develop/src/main/java/edu/harvard/iq/dataverse/util/FileUtil.java#L220

I like the concept of type or class more than group. By calling it a group we don't know what we are grouping based on.

Reply:
Pasting my answer in Slack with a few tweaks, in case someone is interested.

File description as well as other properties of the Datafile or File Metadata models are returned by the main endpoint as they are properties of the models themselves. It is straightforward to include these properties to the json payload from DataFile / FileMetadata instances (See JsonPrinter).

On the other hand, thumbnail class is not a property of any of these models, since it is obtained from the DataFileServiceBean instead. I didn’t really like the idea of including external properties to the main endpoint payload.

And on second thought, thumbnail class is about display logic that I would ideally move to the SPA, but for consistency with JSF, it seems correct that it is obtained from the API (I prefer to leave this display data in an isolated endpoint)

In the future, with only one frontend (The SPA), it may be correct to remove the thumbnail logic from the backend, moving all that part to the SPA instead. Or perhaps, since thumbnail class constants are something that doesn’t change much over time, we could migrate this thumbnail logic to the SPA now and thus avoid adding it to API (With the downside of having this part duplicated on two different sites).

Other consideration that Phil mentioned is that there will probably be multiple frontends that may require the thumbnail class in their display logic, so keeping this logic in the backend would be helpful for them.

I'm still considering the pros and cons for making the right decision.